tools/SourceKit/tools/sourcekitd/bin/XPC/Service/XPCService.cpp

```diff
@@ -208,8 +208,15 @@ std::string sourcekitd::getRuntimeLibPath() {
   return Path;
 }
 
+static void releaseIfMainConnection(xpc_connection_t peer) {
```

Comment:
Is this necessary ? Seems risky to set the global main connection to null. What is the benefit ?

Reply:
Nathan explained the reasoning for this.